Floofbot/Modules/Administration.cs

```diff
@@ -506,6 +510,280 @@ public async Task warnlog([Summary("user")] string user)
             await Context.Channel.SendMessageAsync("", false, builder.Build());
         }
 
+        [Command("forgive", RunMode = RunMode.Async)]
+        [Summary("Remove a user's warning or user notes")]
+        [RequireContext(ContextType.Guild)]
+        [RequireUserPermission(GuildPermission.BanMembers)]
+        public async Task forgiveUser([Summary("warning/usernote")] string type = "", [Summary("user")] string badUser = "")
+        {
+            IQueryable warnings = null;
+            if (string.IsNullOrEmpty(type) || string.IsNullOrEmpty(badUser) || (!type.ToLower().Equals("warning") && !type.ToLower().Equals("usernote"))) // invalid parameters
+            {
+                Embed embed = CreateDescriptionEmbed($"💾 Usage: `forgive [warning/usernote] [user]`");
+                await SendEmbed(embed);
+                return;
+            }
+            IUser user = resolveUser(badUser);
+
+            ulong uID;
+            if (user != null)
+            {
+                uID = user.Id;
+            }
+            else if (Regex.IsMatch(badUser, @"\d{17,18}")) // not in server but valid user id
+            {
+                uID = Convert.ToUInt64(badUser);
+            }
+            else // user not in server AND not valid user id
+            {
+                await Context.Channel.SendMessageAsync($"⚠️ Could not find user \"{badUser}\"");
+                return;
+            }
+
+            if (type == "warning") // forgive warning
+            {
+                if (!_floofDB.Warnings.AsQueryable().Where(w => w.UserId == uID && w.GuildId == Context.Guild.Id && w.Forgiven == false).Any()) // there are no warnings for this user in this guild
+                {
+                    await Context.Channel.SendMessageAsync("User has no warnings to forgive!");
+                    return;
+                }
+                warnings = _floofDB.Warnings.AsQueryable()
+                    .Where(u => u.UserId == uID && u.GuildId == Context.Guild.Id && u.Forgiven == false)
+                    .OrderByDescending(x => x.DateAdded).Take(10);
```

Comment:
Earlier comment was probably lost, but here you `Take(10)` which only returns the ten latest warnings/notes. Is this desired? What if the moderator wants to query for older ones and the user has more than ten?

This same thing is in lines 563, 689, and 700.

Reply:
There will almost never be more than 10 warnings / user notes, however, if there are, we just want to ignore them as thats what was desired in the existing warnlog function and it doesnt seem to make sense to have more warnings in one than the other